### DISCUSSION TITLE
Kohya HRFix

### DISCUSSION THREAD

**Discussion by Kanareika** [lllyasviel/stable-diffusion-webui-forge#235] [General]
I tried to use this extension, but after playing around for a few days, I found that using it seemed to kill creativity. Yes, generation is noticeably faster, but the images become quite identical. I couldn't find a combination of settings that would fix this. I used 1024x1536 resolution - and the results are noticeably less varied, compared to generating at 512x768 and Hires Fix by 2x. I think the problem is most apparent in photorealistic models, but I'm not sure.
Does anyone experience a similar issue?

**Comment by strawberrymelonpanda** [comment]
I'm still trying out different workflows, but it might simply be a case of different tools for different uses.

I tend to toggle it on/off and HyperTile as well.  Sometimes it helps, sometimes it hinders.

**Comment by ArnorWing** [comment]
I wanted to start a discussion on this extension. I tried it a lot with A1111 and got some great results 40% of the time. Lots of mutants. 920x1660 res and yes they looked similar but the detail was amazing. SD1.5
With forge I am using 704x1252 res with settings 1,1.75, .001, 0.3, bislerp, bislerp. From top to bottom. The results are good 80% of the time 
Downscale factor is very sensitive .
Anyone have decent results?

**Reply by zillllis** [reply]
덕분에 좋은 결과를 얻고 있다. 그러나 여러 민감한 요소드이 많이 작용한다.

**Comment by Gothmoth88** [comment]
i wrote this a s a feature request but i think this is more important than new features. imho.
that is why i will post it here again.

-----------------

you do great work but a lot of this work is wasted without proper documentation.
for example the intergrated hiresfix is different to the original one from kohya.
it has different options, i have no clue what the sliders do in the forge implementation.
and that is just one example.

you want to improve webui, to make it better.
an important part to actually make it better is a detailed documentation of what the options do.

i get that writing documentation is not as fun as including new features.
but it is an important part of making a program usable and BETTER.

instead of 20.000 people wasting their time trying to figure stuff out... it would be a better approach if someone who knows this stuff spends some time explaining it.

explaining it in non-coder language.
i don´t need to know the math or process behind every slider in detail.
but at least i need to know what the slider does to the image creation. how to use it for my benefit.

excuse me if i such a documentation exists and i missed it.
i searched but i did not find it. 


----------------


this post shows exactly the problem. 
using these AI tools too often feels like exploring a house with your eyes closed.
it would help to have a basic documentation so you see the doors and don´t have to bump head on into 10 walls before you find a door.
 
how to start finding settings that work when i have NO idea what the sliders do?
sure i can do 300 tests and try to get an idea.... but that´s just a great waste of time.

in the example of the hiresfix... the resulting images change so dramatically that i have no idea what the sliders do even after testing for 2 hours.

i change the "block number" slider from 3 to 4.... and i get a totally different image.
on 3 i have a "head portrait" of a guy screaming... on 4 i have a guy standing surrounded by 5 other guys screaming at him.

downscaling the same. it´s like totally random seeds.


please please..... spend some time on a proper feature documentation.

**Reply by Wontfallo** [reply]
makes ya very curious why someone would go through all the work in implement and intergrade only to fall so simply for lack of docs. I would have thought, at least for me persoinally if I spend so much time and energy building out nerd features like those in forge that I would want to see other's using it

**Comment by GeminiSquishGames** [comment]
Try using the "Extra Settings" that give you a variance seed and amount, use its different resize seed width and height, and also push the starting percent and end percent up to taste. (PS you also may want to keep your seeds the same to get a better idea of what settings are doing, it will at least try to make the same image. Then also have your realtime progress showing and have it update as fast as you can stand the wait time added. That can help you intuit things as they are happening pers step. This can help you figure out what is happening, unfortunately, describing the effects of a value on a car vs. a real life waifu vs. animu wifu is hard to do, as I try to explain in the next paragraphs. Dunno how well...)

It's hard to explain machine learning things even in a coding vernacular. Like how pseudo-random generation works. You may know the seed but you likely won't know the outcome unless you know and understand how the generator takes the seed value and spits out the pattern that seed makes. 

So the problem with knowing how an outside extension or adaptation effects the base model probably starts at the hidden layer of our diffusor/transformer model. These are dependent on how they are trained, and if you are also using adaptations or NNets in the mix with the prompt, it's hard to even visualize in an intuitive, comprehendible way what results you are going to get visually each time. So things are very experimental until someone develops a way to visualize or intuit predictions of the output. If the extension has to inject non-linear algorithms or NNet layers of its own, it's pretty touch and go. So in non-Machine Learning terms, way to much stuff is going on to give a prediction of output that will explain what effect a value has that you are injecting.

**Reply by Wontfallo** [reply]
Because  rand is a lie, just like the cake,  "wait random is actually random?"  🙉  (▀̿Ĺ̯▀̿ ̿)

**Reply by GeminiSquishGames** [reply]
As far as human perceptions are concerned, there are somethings such as random and cake. But in order to make an apple pie... I mean cake from scratch, you must first create the universe... or the simulation. There's still science to be done!
